hedera-mirror-rest/transactions.js

```diff
@@ -320,6 +322,79 @@ const createTransferLists = (rows) => {
   };
 };
 
+/**
+ * Gets consensus_timestamps of the transactions that include a transfer from the staking reward account 800
+ * transfers are found, the staking_reward_transfers property will be added to the associated transaction object.
+ *
+ * @param transferList transfer list
+ * @return {[]|{string}[]}
+ */
+const getStakingRewardTimestamps = (transferList) => {
+  if (!transferList) {
+    return [];
+  }
+
+  return transferList.transactions
+    .map((t) => ({consensus_timestamp: t.consensus_timestamp, transfers: t.transfers}))
+    .filter(
+      (t) =>
+        !_.isEmpty(t.consensus_timestamp) && t.transfers.some((transfer) => transfer.account === STAKING_REWARD_ACCOUNT)
+    )
+    .map((t) => utils.parseTimestampParam(t.consensus_timestamp));
+};
+
+/**
+ * Queries for the staking reward transfer list from the transfer list. If staking reward
+ * transfers are found, the staking_reward_transfers property will be added to the associated transaction object.
+ *
+ * @param transferList transfer list
+ */
+const addStakingRewardTransfers = async (transferList) => {
+  const stakingRewardTimestamps = getStakingRewardTimestamps(transferList);
+  if (stakingRewardTimestamps.length === 0) {
+    return;
+  }
+
+  const positions = _.range(1, stakingRewardTimestamps.length + 1).map((position) => `$${position}`);
+  const whereClause = `where ${StakingRewardTransfer.CONSENSUS_TIMESTAMP} in (${positions.join(',')})`;
+  const query = `select ${StakingRewardTransfer.ACCOUNT_ID},
+                        ${StakingRewardTransfer.AMOUNT},
+                        ${StakingRewardTransfer.CONSENSUS_TIMESTAMP}
+                    from ${StakingRewardTransfer.tableName}
+                    ${whereClause}`;
```

Comment:
also need to add `order by consensus_timestamp asc, account_id asc`, the order for consensus_timestamp should be the same as in the query param, the default is `desc` I believe

```suggestion
  const query = `select ${StakingRewardTransfer.ACCOUNT_ID},
                        ${StakingRewardTransfer.AMOUNT},
                        ${StakingRewardTransfer.CONSENSUS_TIMESTAMP}
                    from ${StakingRewardTransfer.tableName}
                    where ${StakingRewardTransfer.CONSENSUS_TIMESTAMP} in (${positions})`;
```

Reply:
we should change the staking reward transfer query to return a json array which groups by the consensus timestamp, it'll reduce the complexity when merging the staking reward transfers into the transaction rows

Reply:
Awesome thanks. I have made those changes, but I have placed the query results in a map with the consensus timestamp as the key, so there may not be a need to `order by`.